lang/java/reef-wake/wake/src/test/java/org/apache/reef/wake/test/time/util/AlarmProducer.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.wake.test.time.util;
+
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.time.event.Alarm;
+import org.apache.reef.wake.time.runtime.RuntimeClock;
+
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Helper class to generate alarms at user-specified intervals
+ * and count down the barrier on each alarm.
+ */
+public abstract class AlarmProducer implements EventHandler<Alarm> {
```

Comment:
@markusweimer I don't think it can be considered a public API as the class is defined in tests. It won't be even packaged in REEF JAR by default.

The class is abstract because we don't have delegates in Java 1.7. The way it is intended be used is to instantiate objects of this class like this:

```java
      final AlarmProducer alarmProducer = new AlarmProducer(clock, eventCountLatch) {
        @Override
        public int getOffset() {
          return randomOffsetUniform(rand, 1, 100);
        }
      };
```

Also, the class is public to make it visible from unit tests in other packages.